lib/SILOptimizer/Transforms/SILMem2Reg.cpp

```diff
@@ -459,6 +459,7 @@ StackAllocationPromoter::promoteAllocationInBlock(SILBasicBlock *BB) {
   // RunningVal is the current value in the stack location.
   // We don't know the value of the alloca until we find the first store.
   SILValue RunningVal = SILValue();
+  SILValue LastRunningVal = SILValue();
```

Comment:
You should add a comment here explaining what LastRunningVal is and why it is different than RunningVal.

Reply:
You could also change this to be called `PreviousRunningVal` because "last" might mean "the last in a list" whereas previous always means "the one before this one".

Reply:
The information that I would want here is something with the following information (feel free to wordsmith).

```
LastRunningVal is the value that we saw stored in the stack location /immediately before/ we updated the RunningVal at the last store. The reason why we track this information on top of the current running value is that if the last store was an assign and turns out to be dead, we need to insert a destroy_value on that value.
```